### PR TITLE
address_manager: Remove hard-coding of « eth0 » interface

### PR DESCRIPTION
**Pull request by sbadia**
Added a dynamic default interface retrival, using `/proc` in order to no
add an extra dependency (`netifaces`, `pynetinfo`).

This PR, manage both predictable network interface names (`en{s,o,p}`), and old fashion interface  (`eth0`) names.

* https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/
* https://github.com/systemd/systemd/blob/master/src/udev/udev-builtin-net_id.c

Refs: #217

### CODE REVIEW

**Comment by illfelder** [comment]
Hi Sebastien,

Thanks for your contribution! We just completed an overhaul of the Linux guest environment that includes large changes to the address daemon. We are still considering ways to dynamically detect the default interface so I'll try to merge your change into the updated version.

**Comment by sbadia** [comment]
Hi @illfelder ,

Thanks for the comment and informations! I don't know (about the policy for external libs usage), but maybe we can manage this by removing system calls to `ip` and use a common (linux/bsd*/win*) python interface like [netifaces](https://pypi.python.org/pypi/netifaces), in order to solve both issues (default interface #220 / #217 and #110)

wdyt?

Cheers,

**Comment by CrackerJackMack** [comment]
@illfelder Any chance we can see that overhaul in a PR?

**Comment by sbadia** [comment]
Don't know where the patch was commited (seems not a public repository), but it's now fixed in version `1.3.3-0ubuntu4` see https://github.com/GoogleCloudPlatform/compute-image-packages/issues/217#issuecomment-215817565

**Comment by zmarano** [comment]
TL;DR:
Canonical patched their branch to fix Ubuntu 16.04 immediately because it was breaking users. We are working on releasing our rewrite today in a separate branch here (stay tuned). We'll pull in your patch to that branch as well. We won't merge that new branch with master until we are happy with testing however.

**Comment by illfelder** [comment]
The code is now pushed to a development branch of compute image packages:
https://github.com/GoogleCloudPlatform/compute-image-packages/tree/development

**Comment by sbadia** [comment]
Thanks!
